### PR TITLE
ESQL: Add invalid mappings tests with fields from plugins

### PR DESCRIPTION
**Pull request by astefan**
This adds ~~yaml~~ IT tests for:
- checking the Analyzer level checks of incompatible fields usage in queries
- using unsupported field types installed as plugins (mapper-size and mapper-murmur3)

### CODE REVIEW

**Review by nik9000 — APPROVED** [review]

**Comment by astefan** [comment]
@elasticmachine run elasticsearch-ci/part-2

File: x-pack/plugin/esql/qa/server/single-node/build.gradle [hunk 1]
```diff
@@ -21,6 +21,10 @@ restResources {
 tasks.named('javaRestTest') {
   usesDefaultDistribution()
   maxParallelForks = 1
+  dependencies {
```
Comment (mark-vieira) [inline x-pack/plugin/esql/qa/server/single-node/build.gradle:24]: We don't want this block inside the `tasks.named` block. Dependencies should be declared in the root of the build script.

**Review by mark-vieira — CHANGES_REQUESTED** [review]

**Comment by astefan** [comment]
@elasticmachine update branch

File: x-pack/plugin/esql/qa/server/single-node/build.gradle (hunk 1 above)
Comment (astefan) [reply x-pack/plugin/esql/qa/server/single-node/build.gradle]: Done. Thanks @mark-vieira

**Pull request merged**